viz/cmd/tap.go

```diff
@@ -428,15 +460,60 @@ func renderTapEvent(event *tapPb.TapEvent) string {
 }
 
 // renderTapEventJSON renders a Public API TapEvent to a string in JSON format.
-func renderTapEventJSON(event *tapPb.TapEvent) string {
+func renderTapEventJSON(event *tapPb.TapEvent, opts ...renderOptions) string {
+	filter := &renderFilter{}
+	for _, opt := range opts {
+		opt(filter)
+	}
 	m := mapPublicToDisplayTapEvent(event)
+	if filter.JsonPath != "" {
+		filteredJson, err := getJsonFilteredByJPath(m, filter.JsonPath)
+		if err != nil {
+			return err.Error()
```

Comment:
Perhaps I'm a bit pedantic (see my comment on `getJsonFilteredByJPath` ) but it might feel nicer to build JSON encoded strings from errors in this function, especially if client-go's jsonpath util does not already do that for us.

Reply:
@mateiidavid 
Done! Implemented JSON-encoded strings from the errors returned by the client-go.